pkg/resolver/search.go

```diff
@@ -34,10 +35,34 @@ type SearchResult struct {
 
 const ErrorMsg string = "Error building Search query:"
 
+// isContextError checks if an error is a context timeout or cancellation error
+func isContextError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	errMsg := err.Error()
+	if errors.Is(err, context.DeadlineExceeded) ||
+		errors.Is(err, context.Canceled) ||
+		errMsg == "context deadline exceeded" ||
+		errMsg == "context canceled" {
+		klog.V(1).Info(err.Error())
+		return true
+	}
+	return false
+}
+
 func Search(ctx context.Context, input []*model.SearchInput) ([]*SearchResult, error) {
 	defer metrics.SlowLog("SearchResolver", 0)()
 	// For each input, create a SearchResult resolver.
 	srchResult := make([]*SearchResult, len(input))
+
+	select {
+	case <-ctx.Done():
+		return srchResult, errors.New("Timeout processing the request.")
+	default:
+	}
```

Comment:
I don't think this block works as you expect.
It only works if the timeout happens BEFORE you reach the SELECT. If there isn't a timeout, it resolves to `default:` and continues executing the code below. It won't process any timeouts that happen AFTER the SELECT is resolved.